### ISSUE TITLE
Add a checker for superfluous `import a` statements when there is already `import a.b`

### ISSUE THREAD

**Issue by e-gebes** [pylint-dev/pylint#9873]
### Current problem

I was reading the documentation for `ungrouped-imports`:

https://pylint.readthedocs.io/en/stable/user_guide/messages/convention/ungrouped-imports.html

Under "Correct code" it says:

> import logging
> import logging.config

But for me this is not "correct", but bad style because `import logging` is superfluous anyway. `logging` is implicitly imported with `import logging.config`.

See also:

https://docs.python.org/3/reference/simple_stmts.html#the-import-statement ("Examples")
https://docs.python.org/3/reference/import.html#the-module-cache

### Desired solution

I suggest to change the example in the documentation to use `logging.config` and `logging.handlers`.

I also suggest to introduce a checker to warn about such superfluous import statements.

**Comment by akamat10** [comment]
I think that example can be improved. What you are suggesting is covered under `reimport` check.

**Comment by e-gebes** [comment]
> I think that example can be improved. What you are suggesting is covered under `reimported` check.

I tried it out with Pylint 2.17.7 and with 3.3.0, neither one trigger a message on the following piece of code:

```py
import logging
import logging.config
```

**Comment by akamat10** [comment]
Hmm. Need to check it out. I have seen it work recently for a different case. Will investigate.

**Comment by akamat10** [comment]
You are right. I made a mistake. It is not supported. It sounds like a nice feature and is not too difficult to add for the following case:
```
import logging
import logging.config
```

It may potentially need extra work for the reverse order ie.
```
import logging.config
import logging
```
One thing that will need some thought is whether it should fit into an existing warning or whether it should be a new message like say `implicit-reimport`
